pkg/sdk/renderer/argo/renderer.go

```diff
@@ -116,36 +126,51 @@ func (r *Renderer) Render(ctx context.Context, input *RenderInput) (*RenderOutpu
 
 	// 5. List data based on policy and inject them if provided
 	// 5.1 TypeInstances
-	typeInstancesToInject := policyEnforcedClient.ListTypeInstancesToInjectBasedOnPolicy(rule, implementation)
-	err = dedicatedRenderer.InjectDownloadStepForTypeInstancesIfProvided(rootWorkflow, typeInstancesToInject)
+	additionalTypeInstances := policyEnforcedClient.ListTypeInstancesToInjectBasedOnPolicy(rule, implementation)
```

Comment:
This works only with `requires`, so it's a "fake" additionalTypeInstances. Needs to be discussed.

Reply:
Discussed on our engineering call. As a result, we created a dedicated issue to fix it: https://github.com/capactio/capact/issues/438